pyterrier/transformer.py

```diff
@@ -244,52 +262,55 @@ def set_parameter(self, name : str, value):
             raise ValueError(('Invalid parameter name %s for transformer %s. '+
                     'Check the list of available parameters') %(name, str(self)))
 
-    def __call__(self, *args, **kwargs):
+    def __call__(self, *args, **kwargs) -> pd.DataFrame:
         """
             Sets up a default method for every transformer, which is aliased to transform(). 
         """
         return self.transform(*args, **kwargs)
 
-    def __rshift__(self, right):
+    def __rshift__(self, right) -> 'Transformer':
         return ComposedPipeline(self, right)
 
-    def __rrshift__(self, left):
+    def __rrshift__(self, left) -> 'Transformer':
         return ComposedPipeline(left, self)
 
-    def __add__(self, right):
+    def __add__(self, right : 'Transformer') -> 'Transformer':
         return CombSumTransformer(self, right)
 
-    def __pow__(self, right):
+    def __pow__(self, right : 'Transformer') -> 'Transformer':
         return FeatureUnionPipeline(self, right)
 
-    def __mul__(self, rhs):
+    def __mul__(self, rhs : Union[float,int]) -> 'Transformer':
         assert isinstance(rhs, int) or isinstance(rhs, float)
         return ScalarProductTransformer(self, rhs)
 
-    def __rmul__(self, lhs):
+    def __rmul__(self, lhs : Union[float,int]) -> 'Transformer':
         assert isinstance(lhs, int) or isinstance(lhs, float)
         return ScalarProductTransformer(self, lhs)
 
-    def __or__(self, right):
+    def __or__(self, right : 'Transformer') -> 'Transformer':
         return SetUnionTransformer(self, right)
 
-    def __and__(self, right):
+    def __and__(self, right : 'Transformer') -> 'Transformer':
         return SetIntersectionTransformer(self, right)
 
-    def __mod__(self, right):
+    def __mod__(self, right : 'Transformer') -> 'Transformer':
         assert isinstance(right, int)
         return RankCutoffTransformer(self, right)
 
-    def __xor__(self, right):
+    def __xor__(self, right : 'Transformer') -> 'Transformer':
         return ConcatenateTransformer(self, right)
 
-    def __invert__(self):
+    def __invert__(self : 'Transformer') -> 'Transformer':
         from .cache import ChestCacheTransformer
         return ChestCacheTransformer(self)
 
     def __hash__(self):
         return hash(repr(self))
 
+class TransformerBase(Transformer):
```

Comment:
Could this be aliased as `TransformerBase = Transformer` instead? That way, things that inherit from it will be _exactly_ the same as if they inherit from `Transformer`.

(A downside is that it wouldn't allow us to add deprecation warnings to it.)

Maybe a quick comment here too to say that it's to support an old convention.

Reply:
Thanks. I think I'll go for the comment, and then later the deprecation warning in v0.9.